### ISSUE TITLE
Unable to install 

### ISSUE THREAD

**Issue by shekharmittal** [conda/conda#2974]
C:\Users\Chanakya>conda info
Current conda install:

             platform : win-64
        conda version : 4.1.6
    conda-env version : 2.5.1
  conda-build version : 1.14.1
       python version : 2.7.10.final.0
     requests version : 2.10.0
     root environment : C:\Users\Chanakya\Documents\Anaconda2  (writable)
  default environment : C:\Users\Chanakya\Documents\Anaconda2
     envs directories : C:\Users\Chanakya\Documents\Anaconda2\envs
        package cache : C:\Users\Chanakya\Documents\Anaconda2\pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
         offline mode : False
    is foreign system : False

raceback (most recent call last):
  File "C:\Users\Chanakya\Documents\Anaconda2\Scripts\conda-script.py", line 5, in <module>
    sys.exit(main())
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\cli\main.py", line 120, in main
    exit_code = args_func(args, p)
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\cli\main.py", line 130, in args_func
    exit_code = args.func(args, p)
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\cli\main_create.py", line 57, in execute
    install(args, parser, 'create')
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\cli\install.py", line 223, in install
    clone(args.clone, prefix, json=args.json, quiet=args.quiet, index_args=index_args)
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\cli\install.py", line 87, in clone
    index_args=index_args)
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\misc.py", line 360, in clone_env
    force_extract=False, index_args=index_args)
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\misc.py", line 126, in explicit
    index.update(get_index(**index_args))
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\api.py", line 22, in get_index
    channel_urls = normalize_urls(channel_urls, platform, offline)
  File "C:\Users\Chanakya\Documents\Anaconda2\lib\site-packages\conda\config.py", line 250, in normalize_urls
    url = urls[0]
KeyError: 0

There was a problem creating the "dato-env" conda environment.  Restart Dato Launcher.

  If the problem persists, contact support@dato.com with the following:
  Unable to complete creation of conda envinroment "dato-env".

**Comment by mcg1969** [comment]
Can you please offer the exact command you used?

**Comment by shekharmittal** [comment]
Hi Michael

I think I am partly to blame here. I have managed to fix it.

I was doing it through the installation utility of Graphlab.

On my machine, I had Graphlab installed. While cleaning my dropbox, I
accidentally deleted the Anaconda folder.
While trying to reinstall Graphlab, it gave me that error.

How I fixed it:

1. Followed instructions for manual Graphlab installation.
     a. Installed Anaconda-2
     b. Installed Graphlab and then changed the default PATH setting to the
correct new Anaconda path.

Things seem to be working for now. Thanks for reaching out.

Best
Shekhar


On Mon, Jul 4, 2016 at 6:09 PM Michael C. Grant <notifications@github.com>
wrote:

> Can you please offer the exact command you used?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/conda/conda/issues/2974#issuecomment-230281284>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAHk9Qw_aj_mUfouheyezQI6qn20JPvKks5qSP8TgaJpZM4JEQ_A>
> .
>
-- 
http://shekharmittal.info

**Comment by jjhelmus** [comment]
I am running into this same issue. It does not seem as if packages can be install from a file path.

Minimal example:

```
~/sandbox/conda_bug$ conda info
Current conda install:

             platform : linux-64
        conda version : 4.1.6
    conda-env version : 2.5.1
  conda-build version : 1.21.3
       python version : 3.5.1.final.0
     requests version : 2.10.0
     root environment : /home/jhelmus/anaconda  (writable)
  default environment : /home/jhelmus/anaconda/envs/bugenv
     envs directories : /home/jhelmus/anaconda/envs
        package cache : /home/jhelmus/anaconda/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/jhelmus/.condarc
         offline mode : False
    is foreign system : False

~/sandbox/conda_bug$ cat meta.yaml 
package:
  name: foobar
  version: 1.0

~/sandbox/conda_bug$ conda build .
Removing old build environment
BUILD START: foobar-1.0-0
Package: foobar-1.0-0
source tree in: /home/jhelmus/anaconda/conda-bld/work
number of files: 0
Fixing permissions
Fixing permissions
BUILD END: foobar-1.0-0
Nothing to test for: foobar-1.0-0
# If you want to upload this package to anaconda.org later, type:
#
# $ anaconda upload /home/jhelmus/anaconda/conda-bld/linux-64/foobar-1.0-0.tar.bz2
#
# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes

~/sandbox/conda_bug$ conda create -n bugenv python=2.7
Fetching package metadata .......
Solving package specifications: ..........

Package plan for installation in environment /home/jhelmus/anaconda/envs/bugenv:

The following NEW packages will be INSTALLED:

    openssl:  1.0.2h-1 defaults
    python:   2.7.12-1 defaults
    readline: 6.2-2    defaults
    sqlite:   3.13.0-0 defaults
    tk:       8.5.18-0 defaults
    zlib:     1.2.8-3  defaults

Proceed ([y]/n)? y

Linking packages ...
[      COMPLETE      ]|###################################################| 100%
#
# To activate this environment, use:
# $ source activate bugenv
#
# To deactivate this environment, use:
# $ source deactivate
#
~/sandbox/conda_bug$ source activate bugenv
~/sandbox/conda_bug$ conda install /home/jhelmus/anaconda/conda-bld/linux-64/foobar-1.0-0.tar.bz2
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/home/jhelmus/anaconda/envs/bugenv/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/home/jhelmus/anaconda/lib/python3.5/site-packages/conda/cli/main.py", line 120, in main
    exit_code = args_func(args, p)
  File "/home/jhelmus/anaconda/lib/python3.5/site-packages/conda/cli/main.py", line 130, in args_func
    exit_code = args.func(args, p)
  File "/home/jhelmus/anaconda/lib/python3.5/site-packages/conda/cli/main_install.py", line 69, in execute
    install(args, parser, 'install')
  File "/home/jhelmus/anaconda/lib/python3.5/site-packages/conda/cli/install.py", line 203, in install
    explicit(args.packages, prefix, verbose=not args.quiet)
  File "/home/jhelmus/anaconda/lib/python3.5/site-packages/conda/misc.py", line 126, in explicit
    index.update(get_index(**index_args))
  File "/home/jhelmus/anaconda/lib/python3.5/site-packages/conda/api.py", line 22, in get_index
    channel_urls = normalize_urls(channel_urls, platform, offline)
  File "/home/jhelmus/anaconda/lib/python3.5/site-packages/conda/config.py", line 250, in normalize_urls
    url = urls[0]
KeyError: 0
```

**Comment by jjhelmus** [comment]
The package in the example above installs fine with conda version 4.1.5.

**Comment by jjhelmus** [comment]
The **channels** variable is created as a dict in the [`explicit` function in misc.py](https://github.com/conda/conda/blob/4.1.6/conda/misc.py#L41) which is passed to [`get_index` in api.py](https://github.com/conda/conda/blob/4.1.6/conda/api.py#L10) and in turn [`normalize_urls` in config.py](https://github.com/conda/conda/blob/4.1.6/conda/config.py#L245) which is expecting a tuple.  Does **channels** need to be cast to a tuple somewhere along the line?

**Comment by jjhelmus** [comment]
27ab7fc85734b64de9ac9d10ebc71da25012e9a0 seems like a likely cause

**Comment by mcg1969** [comment]
We're working on it, @jjhelmus, sorry for the trouble. Yes you've identified the problem

**Comment by kalefranz** [comment]
fixed with #2976
